File: eCommerce/eCommerce.Services/Database/UserActivity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCommerce.Services.Database
{
    public class UserActivity
    {
        public int UserActivityId { get; set; }
        public int KorisnikId { get; set; }
        public User Korisnik { get; set; }
        public int ActivityId { get; set; }
        public Activity Activity { get; set; }
        public DateTime DateAssigned { get; set; }
        public Status StatusActivity { get; set; }
        public string Note { get; set; }
        public DateTime CompletedAt { get; set; }
        public string? RewardTitle { get; set; }
        public DateTime? RewardedAt { get; set; }
        public enum Status
        {
            Assigned,
            InProgress,
            Completed,
            Cancelled
        }
    }
}

File: eCommerce/UI/ecommerce_mobile/lib/model/useractivity.dart
import 'package:ecommerce_mobile/model/activity.dart';
import 'package:ecommerce_mobile/model/rewardrule.dart';
import 'package:ecommerce_mobile/model/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'useractivity.g.dart';

@JsonSerializable()
class UserActivity {
  int? userActivityId;
  int? korisnikId;
  User? korisnik;
  int? activityId;
  Activity? activity;
  DateTime? dateAssigned;
  String? statusActivity; 
  String? note;
  DateTime? completedAt;
  String? rewardTitle;
  DateTime? rewardedAt;
  RewardRule? rewardRule;

  UserActivity({
    this.userActivityId,
    this.korisnikId,
    this.korisnik,
    this.activityId,
    this.activity,
    this.dateAssigned,
    this.statusActivity,
    this.note,
    this.completedAt,
    this.rewardTitle,
    this.rewardedAt,
  });

  factory UserActivity.fromJson(Map<String, dynamic> json) => _$UserActivityFromJson(json);

  Map<String, dynamic> toJson() => _$UserActivityToJson(this);
}
